runtime/src/main/java/org/corfudb/runtime/view/OrderedGUIDGenerator.java

```diff
@@ -0,0 +1,21 @@
+package org.corfudb.runtime.view;
+
+import java.util.UUID;
+
+/**
+ * Globally Unique Identity (GUID) generator that returns GUIDs having a notion of
+ * a comparable global ordering.
+ *
+ * Created by Sundar Sridharan on 5/22/19.
+ */
+public interface OrderedGUIDGenerator {
+    /**
+     * @return a compact but low resolution ordered guid
+     */
+    long nextLong();
+
+    /**
+     * @return a high resolution ordered guid
+     */
+    UUID nextUUID();
```

Comment:
`nextGuid` ?

Reply:
well, they are both Guids (globally unique identifiers) I used nextUUID() and nextLong() to denote the return type (similar to Java's Random)

Reply:
ok, up to you